packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js

```diff
@@ -58,18 +58,33 @@ angular.module('App').controller(
              ============================== */
       this.loading.availableRam = true;
 
-      this.privateDatabaseService.listAvailableRam().then((availableRam) => {
-        this.loading.availableRam = false;
-        this.data.availableRam = availableRam;
-        if (this.database.infrastructure === 'legacy') {
-          remove(this.data.availableRam, (ram) => ram === '2048');
-        }
-
-        remove(
-          this.data.availableRam,
-          (ram) => +ram === +this.database.ram.value,
-        );
-      });
+      this.privateDatabaseService
+        .getUpgradePlans(this.productId)
+        .then((result) => {
+          this.data.availableRam = result.map(
+            (e) => e.planCode.match(/[0-9]+/)?.[0],
+          );
+          if (this.database.infrastructure === 'legacy') {
+            remove(this.data.availableRam, (ram) => ram === '2048');
```

Comment:
Couldn't this logic be handled by API with available planCode depending on the DB ?

Reply:
I think it is already, but DB team (Mikael Davranche) advised me to keep the piece of code to be on the safe side